core/string/fuzzy_search.cpp

```diff
@@ -172,7 +172,7 @@ Vector<Ref<FuzzySearchResult>> FuzzySearch::sort_and_filter(const Vector<Ref<Fuz
 	};
 
 	// Prune low score entries before sorting
-	for (Ref<FuzzySearchResult> i : p_results) {
+	for (const Ref<FuzzySearchResult>& i : p_results) {
```

Comment:
I think this was creating a new Ref every loop. If you look inside the copy constructor of Ref, it's a little heavy so better to avoid.